src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -1436,6 +1437,18 @@ static LPWSTR LOADGetModuleFileName(MODSTRUCT *module)
     return module->lib_name;
 }
 
+static bool ShouldRedirectToCurrentLibrary(LPCSTR libraryNameOrPath)
+{
+    if (!g_running_in_exe)
+        return false;
+
+    // Getting nullptr as name indicates redirection to current library
+    if (libraryNameOrPath == nullptr)
+        return true;
```

Comment:
They will be linked, as a separate change. Not sure yet if that linking will happen in coreclr or in installer partition. From layering point, and order of the build, probably in installer.

Reply:
Then we will need to add detection of their names here, probably by adding a new property passed to coreclr_initialize with a list of library names that are linked into the binary

Reply:
Yes, even if they will be linked on installer side, we will need to know here what is linked. Not yet sure if hardcoded or names will be passed in. "Passed in" seems more flexible.

Reply:
We are linking System.Native.Globalization into the regular coreclr.dll/libcoreclr.so already. We should make all of these to be on the same plan.

> layering point, and order of the build, probably in installer

FWIW, I think it would be better for the link everything together under coreclr and just reach out to the installer partition for the sources we need. Similar to how we are doing that for System.Native.Globalization today. It would save us from sending large transient artifacts between partitions in the official build.

Reply:
I assumed that libs build after coreclr. If we have a solution for `System.Native.Globalization` to do it in opposite order, I will take a look. It could be simpler.

Reply:
I assume we do not need to do anything about ` System.IO.Compression.Native` and the like in _this_ change?

Reply:
We don't. We will need to update the stuff here once we statically link in those libraries too.

Reply:
Thanks! Yes, this is a good place to shim loading of the native libraries. 
Using `System.Native.Globalization` approach will be my first choice. In such case we will statically know what is embedded, so will not need hints from the host. 
Anyways, that will be the next part.